File: hash/js/Hash-map.js
class HashMap {
  constructor() {
    this.map = {};
  }

  set(key, value) {
    this.map[key] = value;
  }

  get(key) {
    if (this.map.hasOwnProperty(key)) {
      return this.map[key];
    }
    return "not found";
  }
}

const myHashMap = new HashMap();
console.log(myHashMap.set("name", "dipty"));
console.log(myHashMap.get("name"));
console.log(myHashMap);

File: hash/py/Hash-map.py


class Hash_Map:

    def __init__(self):
        self.hash_map = {}


    def __str__(self):
        return f"{self.hash_map}"
    

    def set(self,key,val):
        self.hash_map[key] = val

    
    def get(self,key):

        if key in self.hash_map:
           return self.hash_map[key]
        
        return "not found"

    

my_hash_map = Hash_Map()
print(my_hash_map.set("name","dipty"))
print(my_hash_map.set("age",15))
print(my_hash_map.get("age"))
print(my_hash_map)